From 19941dbbcdbdad7813e45ff605d8c9f5fb2497ac Mon Sep 17 00:00:00 2001
From: Saurav Shah <quic_sauravs@quicinc.com>
Date: Tue, 11 Jun 2024 14:15:34 +0530
Subject: [PATCH] qdl: Modify Makefile

Modify Makefile to compile only qdl

This patch file is from Qualcomm Innovation Center, Inc. and is provided under the following license:

Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
SPDX-License-Identifier: BSD-3-Clause-Clear 

Signed-off-by: Saurav Shah <quic_sauravs@quicinc.com>
---
 0001-qdl-Change-usb-macro.patch | 29 +++++++++++++++++++++++++++++
 Makefile                        |  4 ++--
 2 files changed, 31 insertions(+), 2 deletions(-)
 create mode 100644 0001-qdl-Change-usb-macro.patch

diff --git a/0001-qdl-Change-usb-macro.patch b/0001-qdl-Change-usb-macro.patch
new file mode 100644
index 0000000..1f4a6e7
--- /dev/null
+++ b/0001-qdl-Change-usb-macro.patch
@@ -0,0 +1,29 @@
+From 9a3888b7013d5cca251b0a63585fd211362e9eac Mon Sep 17 00:00:00 2001
+From: Saurav Shah <sauravs@qti.qualcomm.com>
+Date: Mon, 10 Jun 2024 20:46:59 +0530
+Subject: [PATCH] qdl: Change usb macro
+
+Change usb macro to fix compilation
+issues.
+
+Signed-off-by: Saurav Shah <sauravs@qti.qualcomm.com>
+---
+ usb.c | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/usb.c b/usb.c
+index 35a1c1e..fbdce7a 100644
+--- a/usb.c
++++ b/usb.c
+@@ -81,7 +81,7 @@ static int qdl_try_open(libusb_device *dev, struct qdl_device *qdl, const char *
+ 			endpoint = &ifc->endpoint[l];
+ 
+ 			type = endpoint->bmAttributes & LIBUSB_TRANSFER_TYPE_MASK;
+-			if (type != LIBUSB_ENDPOINT_TRANSFER_TYPE_BULK)
++			if (type != LIBUSB_TRANSFER_TYPE_BULK)
+ 				continue;
+ 
+ 			if (endpoint->bEndpointAddress & LIBUSB_ENDPOINT_IN) {
+-- 
+2.17.1
+
diff --git a/Makefile b/Makefile
index 4724858..4869687 100644
--- a/Makefile
+++ b/Makefile
@@ -15,7 +15,7 @@ KS_OUT := ks
 KS_SRCS := ks.c sahara.c util.c
 KS_OBJS := $(KS_SRCS:.c=.o)
 
-default: $(QDL) $(RAMDUMP) $(KS_OUT)
+default: $(QDL)
 
 $(QDL): $(QDL_OBJS)
 	$(CC) -o $@ $^ $(LDFLAGS)
@@ -35,6 +35,6 @@ clean:
 	rm -f $(KS_OUT) $(KS_OBJS)
 	rm -f compile_commands.json
 
-install: $(QDL) $(RAMDUMP) $(KS_OUT)
+install: $(QDL)
 	install -d $(DESTDIR)$(prefix)/bin
 	install -m 755 $^ $(DESTDIR)$(prefix)/bin
-- 
2.17.1

